source/tyr/tyr/resources.py

```diff
@@ -657,10 +680,10 @@ def put(self, name=None, traveler_type=None):
         if profile is None:
             return {'error': 'Non profile is found to update'}, 404
         try:
-            for (attr, default_value) in default_traveler_profile_params[traveler_type].iteritems():
+            for (attr, args_value) in self.args.iteritems():
                 # override hardcoded values by args if args are not None
-                value = default_value if self.args.get(attr) is None else self.args.get(attr)
-                setattr(profile, attr, value)
+                if args_value is not None:
+                    setattr(profile, attr, args_value)
```

Comment:
it's no longer the same...  in "PUT" method, I **update** only attributes that are mentioned by user, on contrary, the old version update all attributes even though those are not  wanted by user... it's a bug...